Storage/tests/System/ManageObjectsTest.php

```diff
@@ -56,6 +56,34 @@ public function testListsObjects()
         $this->assertEquals($objectsToCreate, $foundObjects);
     }
 
+    public function testListsObjectsWithMatchGlob()
+    {
+        $bucket = self::createBucket(self::$client, uniqid('matchglob-'));
+        $objectsToCreate = ["foo/bar", "foo/baz", "foo/foobar", "foobar"];
+        $matchGlobCases = [
+            'foo*bar' => ['foobar'],
+            'foo**bar' => ['foo/bar', 'foo/foobar', 'foobar'],
+            '**/foobar' => ['foo/foobar', 'foobar'],
+            '*/ba[rz]' => ['foo/bar', 'foo/baz'],
+            '*/ba[!a-y]' => ['foo/baz'],
+            '**/{foobar,baz}' => ['foo/baz', 'foo/foobar', 'foobar'],
+            'foo/{foo*,*baz}' => ['foo/baz', 'foo/foobar'],
+        ];
+
+        foreach ($objectsToCreate as $objectToCreate) {
+            $bucket->upload(self::DATA, ['name' => $objectToCreate]);
+        }
+
+        foreach ($matchGlobCases as $matchGlob => $expectedObjects) {
+            $objects = $bucket->objects(['matchGlob' => $matchGlob]);
+            $resultObjects = [];
+            foreach ($objects as $object) {
+                $resultObjects[] = $object->name();
+            }
+            $this->assertEquals($expectedObjects, $resultObjects);
```

Comment:
This test fails when running in PROD. Can you check?
For all cases, `$resultObjects` contains all objects from `$objectsToCreate` which basically means the `matchGlob` param isn't  applied correctly.

Reply:
I re-checked this in my local after merging the service definition upgrade PR and it works fine as expected. The `matchGlob` param provides filtered objects.